src/Wallabag/AnnotationBundle/Controller/WallabagAnnotationController.php

```diff
@@ -86,16 +88,23 @@ public function putAnnotationAction(Annotation $annotation, Request $request)
     {
         $data = json_decode($request->getContent(), true);
 
-        if (!is_null($data['text'])) {
-            $annotation->setText($data['text']);
-        }
+        $form = $this->get('form.factory')->createNamed('', EditAnnotationType::class, $annotation, [
+            'csrf_protection' => false,
+            'allow_extra_fields' => true,
+        ]);
+        $form->submit($data);
 
-        $em = $this->getDoctrine()->getManager();
-        $em->flush();
+        if ($form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($annotation);
+            $em->flush();
 
-        $json = $this->get('serializer')->serialize($annotation, 'json');
+            $json = $this->get('serializer')->serialize($annotation, 'json');
 
-        return (new JsonResponse())->setJson($json);
+            return JsonResponse::fromJsonString($json);
+        }
+
+        return $form;
```

Comment:
Should return a json response

http://symfony.com/doc/current/bundles/FOSRestBundle/2-the-view-layer.html#forms-and-views